### PR TITLE
Add/editing toolkit coming soon

### PR DESCRIPTION
**Pull request by ramonjd**
#### Changes proposed in this Pull Request

This is a PR to try things out. Test our approach.

#### Testing instructions

D45700-code

Fixes #

### CODE REVIEW

**Comment by andrewserong** [comment]
I don't have much to add yet, but this is coming along nicely already! I was pleasantly surprised at how easily this seems to be able to override the existing coming soon mode by just setting public to 0, and therefore none of the existing coming soon behaviour will fire. I think I might have been overthinking things in regards to supporting coming soon v1.

I'm not _too_ sure where you're up to in the PR yet, but I was able to successfully display the fallback coming soon on my test site, but switching a page to Coming Soon from the pages view looks like it's doing the correct thing in Calypso, but doesn't appear to show the page for me yet. Calling `get_blog_option` in my sandbox shows that the `wpcom_coming_soon_page_id` setting hasn't been updated even though the response from the server with D45700-code is applied looks like it _has_... I might be missing something obvious here, so will give it another try tomorrow!

**Comment by ramonjd** [comment]
thanks for testing @andrewserong !!

> Calling get_blog_option in my sandbox shows that the wpcom_coming_soon_page_id setting hasn't been updated even though the response from the server with D45700-code is applied looks like it has... I might be missing something obvious here, so will give it another try tomorrow!

I was just trying to get that working this afternoon when time ran out. Feel free to take a stab tomorrow if you dare ;)

**Comment by andrewserong** [comment]
Thanks @ramonjd! Got it working, just needed a tiny update to D45700-code (the site option was set to `page_for_coming_soon` so I've changed it to `wpcom_coming_soon_page_id` to match this PR). Seems to be working okay, now!

**Comment by andrewserong** [comment]
I thought I might have a go at copying over the markup for the existing Coming Soon page to this PR later on this afternoon — but let me know if I'll be stepping on toes by messing around in this PR and I can branch off this one, or look at a different area (like getting the launch / logged in coming soon banner to support the new public coming soon mode).

**Comment by ramonjd** [comment]
> the site option was set to page_for_coming_soon so I've changed it to wpcom_coming_soon_page_id to match this PR

Awesome, thank you for catching that one I missed!

> I thought I might have a go at copying over the markup for the existing Coming Soon page to this PR later on this afternoon

Go ahead to do whatever you want. Don't wanna break your stride. I'll peel off and look at something else.

**Comment by andrewserong** [comment]
## Fallback coming soon

I've added the current Coming Soon markup as the fallback and got it to load Recolata

* [ ] I notice the current theme is overriding some styling to do with the buttons, which will need to be fixed up on mobile

image

## Coming soon page selected

Here's what it looks like if I pick a Coming Soon page (here I've just created a simple page with a big Cover block and gradient background):

image

## Other screenshots of the current state of the PR

Just adding a couple of other screenshots to show the current state of the PR for anyone else looking on :)

image

image

There's still heaps more to do of course!

**Comment by roo2** [comment]
Amazing progress!
I tested locally, here's what I've found still to do so far

*Settings Page*
* Make unlaunched sites start as coming soon, make launching a site set it to public, not coming soon
* I think the coming soon settings section should have a link to set the coming soon page

*Pages List*
* We should probably only be able to set the coming soon page if the settings is enabled? or otherwise indicate that the page won't be used if the setting is not enabled.
* setting coming soon pages triggers a " Homepage is showing latest posts" message, even if homepage is set to an existing page 🤔 
* once coming soon page is set, the option to set homepage and latest post page disappears from the dropdown menu next to the page  🤔 
* When I had a coming soon page set, as an incognito user, I continued to see the default coming soon page.
* I think sites should start with a default coming soon page (if you're on the right plan?).
* Make the coming soon page feature only work for paid users and communication around that

* Live site *
* When logged in, the v1 coming soon shows a site banner saying "your site is not launched yet, only you can see it until the site is launched" I think we should do something like that too.
* Fix login link on default coming soon page, currently it just goes to #
* a8cs can see peoples coming soon sites, is that expected?

**Comment by ramonjd** [comment]
Thanks for capturing these ideas! 

> Make unlaunched sites start as coming soon, make launching a site set it to public, not coming soon

Maybe we could start on this, putting it behind some query option in `sites/new` and/or feature flag? There's an [issue](https://github.com/Automattic/wp-calypso/issues/45019) that loosely related to this topic.

> I think the coming soon settings section should have a link to set the coming soon page

To the editor version? Either way, good idea. Noted.

> setting coming soon pages triggers a " Homepage is showing latest posts" message, even if homepage is set to an existing page 🤔
> once coming soon page is set, the option to set homepage and latest post page disappears from the dropdown menu next to the page 🤔

This would be worth looking into now too @roo2 

Not sure what the right logic would be, but I guess we should be able to switch between home and coming soon types as we wish. Noting that if a user switches a coming soon page to be the home page, they'll fallback back to the default coming soon page (if the feature is activated.

> a8cs can see peoples coming soon sites, is that expected?

I guess so. I think only the user themselves sees the full site if they're logged in (and with coming soon enabled)


> Fix login link on default coming soon page, currently it just goes to #


Maybe we could address that in https://github.com/Automattic/wp-calypso/pull/44888 ?

**Comment by roo2** [comment]
>> setting coming soon pages triggers a " Homepage is showing latest posts" message, even if homepage is set to an existing page 🤔
>> once coming soon page is set, the option to set homepage and latest post page disappears from the dropdown menu next to the page 🤔
> This would be worth looking into now too @roo2

Great, yea I'd quite like to take a look at this on this PR/branch

**Comment by roo2** [comment]
rebased

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-coming-soon/class-wpcom-coming-soon.php [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+	public static function render_fallback_coming_soon_page() {
+		remove_action( 'wp_enqueue_scripts', 'wpcom_actionbar_enqueue_scripts', 101 );
+		remove_action( 'wp_head', 'print_emoji_detection_script', 7 );
+		remove_action( 'wp_print_styles', 'print_emoji_styles' );
+		remove_action( 'wp_head', 'header_js', 5 );
+		remove_action( 'wp_head', 'global_css', 5 );
+		remove_action( 'wp_footer', 'wpcom_subs_js' );
+		remove_action( 'wp_footer', 'stats_footer', 101 );
+		wp_enqueue_style( 'recoleta-font', '//s1.wp.com/i/fonts/recoleta/css/400.min.css' );
```
Comment (simison) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-coming-soon/class-wpcom-coming-soon.php:174]: No strong opinion on this, but you might also look at importing in CSS [from the typography package](https://github.com/Automattic/wp-calypso/tree/master/packages/typography).

**Comment by lsl** [comment]
Did a bit of work on this in #45581 let me know if all that is ok to push into this, I'm not sure I've done the typography import right so didn't feel too comfortable pushing into this PR directly.

**Review by wp-desktop — CHANGES_REQUESTED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@roo2 please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/886637). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

**Comment by ramonjd** [comment]
Closing this one in favour of https://github.com/Automattic/wp-calypso/pull/45581

**Pull request closed without merging**